### PR TITLE
FocusZone: check for RTL and do correct top to bottom comparisons

### LINKED ISSUE #5438: FocusZone: bidirectional not working correctly in RTL
### Bug Report 
 
- __Package version(s)__: office-ui-fabric-react 6.25.0
- __Browser and OS versions__: Chrome 67.0.3396.99 (Official Build) (64-bit) and IE 11.2312.14393.0
 
 #### Requested priority: High
 
 #### Describe the issue:

On the 'FocusZone' component, setting the 'direction' attribute to 'FocusZoneDirection.bidirectional' does not work as expected for RTL languages (dir="RTL" has been set on the parent 'Fabric' component).

We have a grid of tiles very similar to the grids under 'Variants' here - https://developer.microsoft.com/en-us/fabric#/components/focuszone

Set focus on the first tile by clicking on it (first tile = first row, 'last' col - since this is RTL).
Hit right arrow - Nothing happens (expected since there is nothing on the right)
Hit left arrow - Focus shifts to tile below, rather than the tile on the left (<= BUG).

On the other hand, when you click on some tile in the middle (any tile with tiles on the left and right), then hitting the left arrow key takes you left, and right arrow key takes you right. (works as expected)
 
#### Expected behavior:
 
Setting focus on the first tile (first tile = first row, 'last' col - since this is RTL) and then
hitting 'left' arrow key - should shift focus to the left tile (and not to the tile below)

Just as a side note and if it helps at all - Changing the 'direction' to 'FocusZoneDirection.horizontal' fixes this (left arrow key) issue and displays the expected behavior for left and right arrow keys (but this option does not support vertical navigation - which is expected). 

But our use case requires both horizontal and vertical navigation to work and hence the bug report on 'FocusZoneDirection.bidirectional'.

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #5438 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

This PR changes one of the comparisons in `_moveFocusLeft` and `_moveFocusRight` when in RTL.
When moving focus left in RTL, instead of comparing the target rect bottom with the active rect top, we should compare the target rect top with the active rect bottom. Similarly, when moving focus right in RTL, instead of comparing the target rect top with the active rect bottom, we should compare the target rect bottom with the active rect top. 

Before (hitting the left arrow key): 
focuszone bidirectional rtl not working

After (hitting the left arrow key):
focuszone rtl bidirectional working

Here's my reasoning, let me know if it makes sense.

RTL is different than LTR in this scenario in a couple ways. For one, going right at a rightmost rectangle will go up a line instead of down a line like in LTR. Similarly, going left at a leftmost rectangle will go down a line instead of up a line like in LTR. This is important for comparisons, because the `_moveFocusLeft` function, when triggered on a rightmost element, will compare everything to the left of the active rectangle, as well as the rectangle immediately below it. Vice versa for `_moveFocusRight`.

LTR:
Moving right
image

Moving left
image

RTL (what we have now):
Moving right
image

Moving left
image

RTL (what we will have with these changes):
Moving right
image

Moving left
image

I also made a change to round the bottom and top comparisons to the 3rd decimal place, so that small rounding errors will not affect these comparisons and move focus down a line when it should just be moved to the left or to the right.

### CODE REVIEW

**Comment by natalieethell** [comment]
To elaborate on the correction for the rounding errors, here's comments I left on the original issue:

It looks like the problem area is in the first comparison here in `_moveFocusRight` for LTR: 
https://github.com/OfficeDev/office-ui-fabric-react/blob/d2778eb85f78a056e552fda52cda17c6bc273d5b/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L697-L702

and in the first comparison here in `_moveFocusLeft` for RTL:
https://github.com/OfficeDev/office-ui-fabric-react/blob/d2778eb85f78a056e552fda52cda17c6bc273d5b/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L664-L669

Sometimes, the `targetRect.top` and `activeRect.bottom` aren't exactly equal to each other for leftmost elements in LTR and rightmost elements in RTL (which they should be), which is causing the code to store this value into distance here for LTR:
https://github.com/OfficeDev/office-ui-fabric-react/blob/d2778eb85f78a056e552fda52cda17c6bc273d5b/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L702

and here for RTL:
https://github.com/OfficeDev/office-ui-fabric-react/blob/d2778eb85f78a056e552fda52cda17c6bc273d5b/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L669

This value will always be 0 because `targetRect.left` and `activeRect.left` will be the same for all the leftmost elements in LTR. Same for `targetRect.right` and `activeRect.right` for all the rightmost elements in RTL. This value is smaller than the value stored for the element horizontally next to the active element.

**Comment by KevinTCoughlin** [comment]
Kudos for adding unit tests 🎉 👍

File: packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.test.tsx [hunk 1]
```diff
@@ -396,7 +396,123 @@ describe('FocusZone', () => {
...
+    ReactTestUtils.Simulate.keyDown(focusZone, { which: KeyCodes.up });
+    expect(lastFocusedElement).toBe(buttonC);
+
+    // Pressing up should go to a.
+    ReactTestUtils.Simulate.keyDown(focusZone, { which: KeyCodes.up });
+    expect(lastFocusedElement).toBe(buttonA);
+  });
+
   it('can use arrows bidirectionally with data-no-vertical-wrap', () => {
+    setRTL(false);
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.test.tsx]: Consider moving this to the end of the test you added

**Comment by cliffkoh** [comment]
I've been out of action the last few days and am behind trying to catch up (and so cannot afford the time to review this PR) but kudos for what looks like amazingly detailed documentation, for tests, and for this fix.

File: packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.test.tsx [hunk 2]
```diff
@@ -396,6 +396,123 @@ describe('FocusZone', () => {
...
+    const buttonD = focusZone.querySelector('.d') as HTMLElement;
+    const buttonE = focusZone.querySelector('.e') as HTMLElement;
+
+    // Set up a grid like so:
+    // B A
+    // hiddenButton C
+    // E D
+    //
+    // We will iterate from A to B, press down to skip hidden and go to C,
+    // down again to E, right to D, then back up to A.
```
Comment (kkjeer) [inline packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.test.tsx]: Nit: in the tests below, it seems like there is another test that goes up from D to C before going back up to A.

**Comment by KevinTCoughlin** [comment]
Unrelated to this change, but one oddity I observed in both `master` and this branch is if you press `UP` to navigate horizontally in a row, say the first row. And then after a few presses within that row you press `DOWN` it jumps to the box that was immediately below where you begin (prior to pressing `UP`).

**Pull request merged**